src/client/v2/algod/models/types.ts

```diff
@@ -688,7 +688,7 @@ export class AssetParams extends BaseModel {
    * (am) A commitment to some unspecified asset metadata. The format of this
    * metadata is up to the application.
    */
-  public metadataHash?: Uint8Array;
+  public metadataHash?: Uint8Array | string;
```

Comment:
What is the purpose of this change? Recently we changed the constructor of this type to allow `metadataHash` (and all other binary data fields) to accept a base64 string as well: https://github.com/algorand/js-algorand-sdk/blob/63c657018a112ad0f9a7e105e82b5157d96ed636/src/client/v2/algod/models/types.ts#L784

This file is actually generated from algod's REST spec, so it would be difficult to change the type of this specific field anyway.

Reply:
we are using `AssetParams` type in runtime in which user can pass string type in metadatahash, it seems it will be difficult to change here, so we will add the conversion using recent changes`Recently we changed the constructor of this type to allow metadataHash (and all other binary data fields) to accept a base64 string`.